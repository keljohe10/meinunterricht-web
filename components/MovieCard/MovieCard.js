import React from 'react';
import PropTypes from 'prop-types';
import Image from 'next/image';
import {
  Box,
  Flex,
  Center,
  Heading,
  Text,
  Stack,
  Tooltip,
} from '@chakra-ui/react';
import { FilmIcon } from 'chakra-ui-ionicons';

const MovieCard = ({ poster, title, plot, director }) => {
  const renderPoster = image => {
    if (image !== 'N/A')
      return (
        <Image
          data-testid='poster'
          src={image}
          alt='movie'
          w={6}
          h={6}
          layout='fill'
        />
      );

    return (
      <Flex
        data-testid='nofound-poster'
        alignItems='center'
        w='full'
        h='full'
        justify='center'
      >
        <FilmIcon w={20} h={20} color='gray.500' />
      </Flex>
    );
  };

  return (
    <Center py={16} mx={2}>
      <Box
        w={{ base: '230px', md: '360px' }}
        h={{ md: '430px' }}
        bg='white'
        boxShadow={'2xl'}
        rounded={'md'}
        p={6}
        overflow={'hidden'}
      >
        <Box
          h={'210px'}
          bg={'gray.100'}
          mt={-6}
          mx={-6}
          mb={6}
          pos={'relative'}
        >
          {renderPoster(poster)}
        </Box>
        <Stack>
          <Heading
            color='gray.700'
            fontSize={{ base: 'md', md: '2xl' }}
            fontFamily={'body'}
          >
            {title}
          </Heading>
          <Tooltip label={plot} aria-label={title} hasArrow>
            <Text
              color={'gray.500'}
              noOfLines={2}
              display={{ base: 'none', md: '-webkit-box' }}
            >
              Plot: {plot}
            </Text>
          </Tooltip>
        </Stack>
        <Stack mt={6} direction={'row'} spacing={4} align={'center'}>
          <Stack direction={'column'} spacing={0} fontSize={'sm'}>
            <Text fontWeight={600}>{director}</Text>
            <Text color={'gray.500'}>2001</Text>
          </Stack>
        </Stack>
      </Box>
    </Center>
  );
};

MovieCard.propTypes = {
  title: PropTypes.string.isRequired,
  plot: PropTypes.string.isRequired,
  director: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
};

export default MovieCard;
