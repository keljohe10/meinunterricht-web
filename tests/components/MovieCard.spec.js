/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';

import { render, cleanup } from '@/tests/testUtils';

import MovieCard from '@/components/MovieCard/MovieCard';

beforeEach(cleanup);

describe('MovieCard component', () => {
  let wrapper;

  const initialValues = {
    title: 'Race to Space',
    director: 'Sean McNamara',
    plot: 'In the 1960s a young woman works at NASA as an animal trainer responsible for the chimpanzee who will go into space.',
    poster:
      'https://m.media-amazon.com/images/M/MV5BMTczNDUwMDA1NV5BMl5BanBnXkFtZTcwNTA5MTAwMQ@@._V1_SX300.jpg',
  };

  const renderComponent = props => render(<MovieCard {...props} />);

  it('matches snapshot', () => {
    const component = renderComponent(initialValues);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should render movie information', () => {
    const component = renderComponent(initialValues);

    expect(component.getByText(initialValues.title)).toBeInTheDocument();
    expect(component.getByText(initialValues.director)).toBeInTheDocument();
    expect(component.getByTestId('poster')).toBeInTheDocument();
  });

  it('should render a movie icon when there is no a poster', () => {
    const component = renderComponent({ ...initialValues, poster: 'N/A' });

    expect(component.getByTestId('nofound-poster')).toBeInTheDocument();
  });
});
