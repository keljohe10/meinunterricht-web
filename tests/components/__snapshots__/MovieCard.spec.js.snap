// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1grfepy"
  >
    <div
      class="css-13rp7ie"
    >
      <div
        class="css-1yzea99"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="movie"
            data-nimg="fill"
            data-testid="poster"
            decoding="async"
            h="6"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            w="6"
          />
          <noscript />
        </span>
      </div>
      <div
        class="chakra-stack css-n21gh5"
      >
        <h2
          class="chakra-heading css-1f7s196"
        >
          Race to Space
        </h2>
        <p
          class="chakra-text css-14qowsz"
        >
          Plot: In the 1960s a young woman works at NASA as an animal trainer responsible for the chimpanzee who will go into space.
        </p>
      </div>
      <div
        class="chakra-stack css-v0cdwd"
      >
        <div
          class="chakra-stack css-1r2fu7y"
        >
          <p
            class="chakra-text css-35ezg3"
          >
            Sean McNamara
          </p>
          <p
            class="chakra-text css-ax9bdp"
          >
            2001
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
